google/cloud/storage/internal/rest/stub.cc

```diff
@@ -106,6 +106,36 @@ StatusOr<ReturnType> CreateFromJson(
   return ReturnType::CreateFromJson(*payload);
 }
 
+Status AddCustomHeaders(Options const& options,
+  RestRequestBuilder& builder) {
+    // In tests this option may not be set. And over time we want to retire it.
+    if (!options.has<CustomHeadersOption>()) return {};
+    auto custom_headers =
+        options.get<CustomHeadersOption>();
+    for (auto const& h : options.get<CustomHeadersOption>()) {
+      builder.AddHeader(h.first, h.second);
+    }
+    return {};
+}
+
+void AddOptionsToRequestBuilder(Options const& options,
+  RestRequestBuilder& builder) {
+    Status cust = AddCustomHeaders(options, builder);
```

Comment:
If we don't care about status, then remove it. Otherwise, we will want to add a condition or return it to its caller.